src/NServiceBus.Persistence.CosmosDB/TransactionalBatchDecoratorExtensions.cs

```diff
@@ -2,43 +2,58 @@
 {
     using System;
     using System.Collections.Generic;
-    using System.Net;
     using System.Threading.Tasks;
 
     static class TransactionalBatchDecoratorExtensions
     {
+        internal static async Task Execute(this TransactionalBatchDecorator transactionalBatch, Operation operation)
+        {
+            operation.Apply(transactionalBatch);
+
+            using (var batchOutcomeResponse = await transactionalBatch.Inner.ExecuteAsync().ConfigureAwait(false))
+            {
+                if (batchOutcomeResponse.Count > 1)
+                {
+                    throw new Exception("The transactional batch can only have one operation.");
```

Comment:
Do I understand correctly that this is very specific for the outbox dispatch case (`SetAsDispatched`)? I.e. no other operations would be expected to be found other than the outbox operation. If that's the case, the exception could be more informative and elaborate on why we're throwing.

Reply:
Also, is it possible to get into this scenario other than us using the API incorrectly?

Reply:
That is how I understand it. Only we would cause this situation.

Reply:
From the perspective of this code this exception enforces that this method can only ever have a single operation within a transactional batch, nothing prevents this method from being used elsewhere as long as these requirements are fulfilled. That's why I did not mention directly the outbox part

Reply:
Right, what confused me is the wording.
> The transactional batch can only have one operation.

I would be OK if it's used not just with the Outbox but have the following message:
> The transactional batch was intended to be used with a single operation but contained more than one.

Reply:
Raised https://github.com/Particular/NServiceBus.Persistence.CosmosDB/pull/21